src/tools/interop/idt/capture/controller.py

```diff
@@ -0,0 +1,179 @@
+logger = log.get_logger(__file__)
+
+
+def get_timeout():
+    return asyncio.get_running_loop().time() + config.orchestrator_async_step_timeout
+
+
+class PlatformFactory:
+
+    @staticmethod
+    def list_available_platforms() -> typing.List[str]:
+        return copy.deepcopy(capture.platform.__all__)
+
+    @staticmethod
+    async def get_platform_impl(
+            platform: str,
+            artifact_dir: str) -> PlatformLogStreamer:
+        if platform in _PLATFORM_MAP:
+            return _PLATFORM_MAP[platform]
+        border_print(f"Initializing platform {platform}")
+        platform_class = getattr(capture.platform, platform)
+        platform_artifact_dir = os.path.join(artifact_dir, platform)
+        safe_mkdir(platform_artifact_dir)
+        platform_inst = platform_class(platform_artifact_dir)
+        _PLATFORM_MAP[platform] = platform_inst
+        async with asyncio.timeout_at(get_timeout()):
+            await platform_inst.connect()
+        return platform_inst
+
+
+class EcosystemFactory:
+
+    @staticmethod
+    def list_available_ecosystems() -> typing.List[str]:
+        return copy.deepcopy(capture.ecosystem.__all__)
+
+    @staticmethod
+    async def get_ecosystem_impl(
+            ecosystem: str,
+            platform: PlatformLogStreamer,
+            artifact_dir: str) -> EcosystemCapture:
+        if ecosystem in _ECOSYSTEM_MAP:
+            return _ECOSYSTEM_MAP[ecosystem]
+        ecosystem_class = getattr(capture.ecosystem, ecosystem)
+        ecosystem_artifact_dir = os.path.join(artifact_dir, ecosystem)
+        safe_mkdir(ecosystem_artifact_dir)
+        ecosystem_instance = ecosystem_class(platform, ecosystem_artifact_dir)
+        _ECOSYSTEM_MAP[ecosystem] = ecosystem_instance
+        return ecosystem_instance
+
+    @staticmethod
+    async def init_ecosystems(platform, ecosystem, artifact_dir):
+        platform = await PlatformFactory.get_platform_impl(platform, artifact_dir)
+        ecosystems_to_load = EcosystemFactory.list_available_ecosystems() \
+            if ecosystem == 'ALL' \
+            else [ecosystem]
+        for ecosystem in ecosystems_to_load:
+            try:
+                await EcosystemFactory.get_ecosystem_impl(
+                    ecosystem, platform, artifact_dir)
+            except UnsupportedCapturePlatformException as e:
+                logger.error(f"Unsupported platform {ecosystem} {platform}")
+                track_error(ecosystem, "UNSUPPORTED_PLATFORM", str(e))
+            except Exception as e:
+                logger.error(f"Unknown error instantiating ecosystem {ecosystem} {platform}")
+                track_error(ecosystem, "UNEXPECTED", str(e))
+
+
+def track_error(ecosystem: str, error_type: str, error_message: str) -> None:
+    if ecosystem not in _ERROR_REPORT:
+        _ERROR_REPORT[ecosystem] = []
+    e = traceback.format_exc()
+    logger.error(e)
+    _ERROR_REPORT[ecosystem].append((error_type, error_message, e))
+
+
+class Orchestrator:
```

Comment:
some comments or docstrings about why these classes exist and their purpose would be useful.

Reply:
Why do we have a class with everything @static_method? Should this be a module then instead like `orchestrator.py` or similar?

Reply:
same comment about static-only classes throughout.

Reply:
Fair point. I think an older structure evolved over time and this wasn't serving any purpose. Moved everything to module scope and left the `controller` name.